core/database/database_factory.py

```
@@ -67,17 +69,21 @@ def get_arango(cls,
                 )
                 logger.info("✓ Using Unix socket for ArangoDB connection")
                 return db
-            except ImportError:
-                logger.warning("Unix socket client not available, falling back to HTTP")
+            except ImportError as e:
+                logger.error("Unix socket client not available - this is required for HADES")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)